### DISCUSSION TITLE
How to reset a toggle when another atom is changed?

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#1353] [Q&A] [answered]
How would I reset a toggle created with the `atomWithToggle` function when another boolean atom is changed?

I'd prefer not to use `useEffect` if possible.

**Comment by dai-shi** [comment, marked as answer]
Basically, atoms are defined declaratively (precisely, `read` is declarative, and `write` is imperative.)
When you say, change something based on something, is it declarative or imperative?

**Comment by AjaxSolutions** [comment]
This is good info, thx.
